src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -586,7 +586,7 @@ protected GraphQLFieldDefinition getFieldDef(GraphQLSchema schema, GraphQLObject
             return TypeNameMetaFieldDef;
         }
 
-        GraphQLFieldDefinition fieldDefinition = parentType.getFieldDefinition(field.getName());
+        GraphQLFieldDefinition fieldDefinition = schema.getFieldVisibility().getFieldDefinition(parentType, field.getName());
         if (fieldDefinition == null) {
```

Comment:
This should not be possible if validation catches the unknown field (which we have tests for) but belts and braces